app/gradescope_to_spreadsheet.py

```diff
@@ -0,0 +1,334 @@
+# Author: Naveen Nathan
+
+"""
+This file is deprecated. Please use 'api/app.py' instead.
+"""
+
+import json
+
+from fullGSapi.api import client
+import os.path
+import sys
+import re
+import pandas as pd
+import io
+import time
+import warnings
+import functools
+
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+# CS10 Fa24 course id
+COURSE_ID = "831412"
+# This scope allows for write access.
+SCOPES = ["https://www.googleapis.com/auth/spreadsheets"]
+SPREADSHEET_ID = "1Ms0PvwUKbph3p29bih-rDBV9EjyI9Ae-T3rGlRDwXqc"
+NUMBER_OF_STUDENTS = 187
+# Lab number of labs that are not graded.
+UNGRADED_LABS = [0]
+# Used only for Final grade calculation; not for display in the middle of the semester
+TOTAL_LAB_POINTS = 80
+NUM_LECTURES = 24
+
+# Used for labs with 4 parts (very uncommon)
+SPECIAL_CASE_LABS = [16]
+NUM_LECTURE_DROPS = 3
+
+# The ASSIGNMENT_ID constant is for users who wish to generate a sub-sheet (not update the dashboard) for one assignment, passing it as a parameter.
+ASSIGNMENT_ID = (len(sys.argv) > 1) and sys.argv[1]
+
+# This is not a constant; it is a variable that needs global scope. It should not be modified by the user
+subsheet_titles_to_ids = None
+
+
+
+def deprecated(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        warnings.warn(
+            f"'{func.__name__}' is deprecated and will be removed in a future version.",
+            DeprecationWarning,
+            stacklevel=2
+        )
+        return func(*args, **kwargs)
+    return wrapper
+
+"""
+Allows the user authenticate their google account, allowing the script to modify spreadsheets in their name.
+Borrowed from here: https://developers.google.com/sheets/api/quickstart/python
+"""
+
+@deprecated
+def allow_user_to_authenticate_google_account():
```

Comment:
This whole function needs to be reworked. This library is depecrated 
https://google-auth.readthedocs.io/en/master/oauth2client-deprecation.html

There is a required manual intervention from a user: If no valid credentials are available and refreshing isn’t possible,
`flow.run_local_server(port=0)` opens a local server on an available port and a browser window, prompting the user to sign in to their Google account and grant the specified permissions.

Please work with Connor on creating a service account for this application. This is critical for deployment and automated cron jobs.
https://developers.google.com/identity/protocols/oauth2/service-account

Our application should not require manual intervention for authentication. GradeView also currently uses a service account and we probably need to make our own service account on GCP.

Reply:
See https://github.com/AFA-Tooling/GradeView/blob/main/dbcron/update_bins.py for the service account approach

Reply:
In the `.env` file, we should have a variable `SERVICE_ACCOUNT_CREDENTIALS` with the google service account credentials. We will no longer need token.json

Reply:
I DMed Victoria and Naveen the SERVICE_ACCOUNT_CREDENTIALS are created the service account for GradeSync on GCP